### DISCUSSION TITLE
brownout timeline for the public v7 api (api.cobalt.tools/api/json)

### DISCUSSION THREAD

**Discussion by wukko** [imputnet/cobalt#876] [announcements]
as mentioned in [the previous announcement](https://github.com/imputnet/cobalt/discussions/860), we've decided to shut down public v7 api in order to keep cobalt available for everyone.

we really wish it didn't have to be this way, but it became nearly impossible to provide free & unrestricted api access due to repeated abuse & reselling (wrapping our api in paid services or services with ads).

it will be shut down on **november 11th, 2024**.

before complete unavailability, we will start showing a deprecation message with a link to the initial announcement as an error response to a set % of all traffic.

this will start on november 2nd, with continuous growth until november 11th:
- november 2nd: 10% of traffic
- november 4th: 40% of traffic
- november 9th: 80% of traffic
- november 11th: 100% of traffic

if v10 ever becomes available for public access just like v7, we will post an announcement here!

[cobalt.tools](https://cobalt.tools) isn't affected and will keep working for the foreseeable future, nothing about its availability will change.

**Comment by v1ctorio** [comment]
It was too good to survive _the enterprise_ :(

**Reply by stevoblevo** [reply]
Is this good info? 

### ChatGPT4o says:

To continue using the **YouTube Downloader** userscript beyond **November 11, 2024**, you'll need to modify it to work with the updated **Cobalt API**, as the v7 API has been deprecated. Here's how you can do this:

---

#### 1. Access the Script:
- The original script is available on GitHub: [madkarmaa/youtube-downloader](https://github.com/madkarmaa/youtube-downloader).

---

#### 2. Install Tampermonkey:
- Ensure you have the **Tampermonkey extension** installed in your browser. You can download it from [Tampermonkey's official website](https://www.tampermonkey.net/).

---

#### 3. Modify the Script:
   - **Download the Script**: Navigate to the GitHub repository and download the `script.user.js` file.
   - **Edit the Script**: Open the script in a text editor. Locate any references to the old API endpoint (e.g., `api.cobalt.tools/api/json`) and update them to the new API endpoint as specified in the [Cobalt API repository](https://github.com/cobaltgit/CobaltAPI).
   - **Update API Keys**: If the new API requires authentication, ensure you include the necessary **API keys or tokens** in the script.

---

#### 4. Install the Modified Script:
   - After editing, save the script with a `.user.js` extension.
   - Open Tampermonkey in your browser, click on the **Dashboard**, and select **"Add a new script."**
   - Paste your modified script into the editor and **save it**.

---

#### 5. Test the Script:
   - Navigate to YouTube and test the downloader to ensure it's functioning correctly with the new API.

---

> **Note**: If you're not comfortable modifying the script yourself, consider reaching out to the community or the original script author for an updated version. Additionally, always ensure you're complying with YouTube's terms of service when downloading content.

---
I will test myself in time. Cheers!

**Comment by nichind** [comment]
rip

**Reply by wukko** [reply]
only old api is shutting down, cobalt.tools and new api will stay online!

**Reply by nichind** [reply]
yeah ik
лю тебя wukko

**Comment by axellse** [comment]
sucks now that most invidious instances have stopped working

**Reply by wukko** [reply]
you can host your own cobalt instance, it's very shrimple

**Comment by JumperOnJava** [comment]
can i selfhost it?

**Reply by KwiatekMiki** [reply]
yes

**Comment by faustyu1** [comment]
bruh

**Reply by wukko** [reply]
?
